superset/views/core.py

```diff
@@ -1974,7 +1974,8 @@ def _set_dash_metadata(dashboard, data):
             ):
                 slice_id = value.get("meta").get("chartId")
                 slice_ids.append(slice_id)
-                slice_id_to_name[slice_id] = value.get("meta").get("sliceName")
+                if "sliceName" in value.get("meta"):
```

Comment:
Somewhat unrelated to your change but I sense we can cleanup some of this logic (which could be error prone as `get()` returns `None` if the key doesn't exist), 

```
if (
     isinstance(value, dict)
     and "meta" in value 
     and "chartId" in value["meta"]
):
    slice_id = value["meta"]["chartId"]
    slice_ids.append(slice_id)
    if "sliceName" in value["meta"]:
        slice_id_to_name[slice_id] = value["meta"]["sliceName"]    
```

or 

```
try:
    slice_id = value["meta"]["chartId"]
    ...
except KeyError:
    pass
```

Note I'm uncertain whether `sliceId` or `sliceName` could be `None` and whether additional checks are required.

Reply:
sliceName could be None, which means user doesn't have to **show** a slice with name in dashboard, but the associated Slice entry in the database still have name column

Reply:
fixed.